dialects/mysql/src/main/kotlin/app/cash/sqldelight/dialects/mysql/MySqlType.kt

```diff
@@ -61,9 +61,10 @@ internal enum class MySqlType(override val javaType: TypeName) : DialectType {
     return CodeBlock.of(
       when (this) {
         TINY_INT, TINY_INT_BOOL, SMALL_INT, INTEGER, BIG_INT, BIT -> "$cursorName.getLong($columnIndex)"
-        DATE, TIME, DATETIME, TIMESTAMP -> "$cursorName.getObject($columnIndex)"
+        DATE, TIME, DATETIME, TIMESTAMP -> "$cursorName.getObject<%T>($columnIndex)"
         NUMERIC -> "$cursorName.getBigDecimal($columnIndex)"
       },
+      javaType,
```

Comment:
I suppose `CodeBlock.of()` is fine with extra parameters if they never get used - if tests pass its probably fine